scripts/test-truncating-of-client-hello.py

```diff
@@ -136,66 +133,86 @@ def main():
         # we expect Alert and Close or just Close
         node = node.add_child(ExpectAlert())
         node.next_sibling = ExpectClose()
         node.add_child(ExpectClose())
+        conversations.append(conversation)
 
-        if "pad" in name:
-            conversations[name + ": " + str(pad_len) + " of \"" + str(pad_byte) +
-                          "\" byte padding"] = conversation
+    # first and last test is sanity
+    conversations.insert(0, conversation_sanity)
+    conversations.append(conversation_sanity)
+
+    # TBD: move it to separate function
+    for conversation in conversations:
+        if run_only and conversation.name not in run_only or conversation.name in run_exlude:
+            conversations.remove(conversation)
+
+    run_all(conversations)
+
+def all_passed(conversations):
+    for conversation in conversations:
+        if not conversation.passed:
+            return False
+
+    return True
+
+def print_result(conversations):
+    print("List of Succesfull Tests:")
+    for conversation in conversations:
+        if conversation.passed:
+            print("{0}".format(conversation.name))
+            print("OK\n")
+    print("=================\n")
+    print("List of Failed Tests:\n")
+    for conversation in conversations:
+        if not conversation.passed:
+            print("{0}".format(conversation.name))
+            print("FAILED\n")
+    print("=================\n")
+
+def print_statistic(conversations):
+    passed = 0
+    not_passed = 0
+
+    for conversation in conversations:
+        if  conversation.passed:
+            passed += 1
         else:
-            conversations[name + ": " + str(-pad_len) +
-                          " of bytes truncated"] = conversation
-
-    # run the conversation
-    good = 0
-    bad = 0
-    failed = []
-
-    # make sure that sanity test is run first and last
-    # to verify that server was running and kept running throught
-    sanity_test = ('sanity', conversations['sanity'])
-    ordered_tests = chain([sanity_test],
-                          filter(lambda x: x[0] != 'sanity',
-                                 conversations.items()),
-                          [sanity_test])
-
-    for c_name, c_test in ordered_tests:
-        if run_only and c_name not in run_only or c_name in run_exclude:
-            continue
-        print("{0} ...".format(c_name))
-
-        runner = Runner(c_test)
-
-        res = True
+            not_passed += 1
+
+    print("Test end")
+    print("successful: {0}".format(passed))
+    print("failed: {0}".format(not_passed))
+
+def run_all(conversations):
+    for conversation in conversations:
+        runner = Runner(conversation.root_node)
+
         try:
             runner.run()
         except:
-            print("Error while processing")
-            print(traceback.format_exc())
-            res = False
-
-        if res:
-            good += 1
-            print("OK\n")
+            # TBD: print(traceback.format_exc())
```

Comment:
the traceback is the only thing that contains information when the test fails, it is quite important...